### PR TITLE
Prefactor file download

### PR DESCRIPTION
**Pull request by nickpell**
@dannyroberts Started this to make https://github.com/dimagi/commcare-hq/pull/9938 more testable.  After getting started I'm not sure if it's more easily testable than before, but I still think the refactoring is an improvement.

### CODE REVIEW

File: corehq/apps/app_manager/views/__init__.py [hunk 1]
```diff
@@ -66,9 +41,15 @@
     download_test_jar,
     download_user_registration,
     download_xform,
+    DownloadBuildAttachmentsView,
     DownloadCCZ,
     validate_form_for_build,
 )
+from corehq.apps.app_manager.views.formdesigner import (
```
Comment (dimagimon) [inline corehq/apps/app_manager/views/__init__.py]: F401 'form_designer' imported but unused

Comment (dimagimon) [inline corehq/apps/app_manager/views/__init__.py]: F401 'get_data_schema' imported but unused

Comment (dimagimon) [inline corehq/apps/app_manager/views/__init__.py]: F401 'user_registration_source' imported but unused

File: corehq/apps/app_manager/views/download.py [hunk 2]
```diff
@@ -9,15 +9,19 @@
...
 
 from corehq.apps.app_manager.dbaccessors import get_app
 from corehq.apps.app_manager.decorators import safe_download
 from corehq.apps.app_manager.exceptions import ModuleNotFoundException, \
     AppManagerException, FormNotFoundException
 from corehq.apps.app_manager.util import add_odk_profile_after_build
+from corehq.apps.app_manager.view_helpers import ApplicationViewMixin
 from corehq.apps.app_manager.views.utils import back_to_main, get_langs
 from corehq.apps.hqmedia.views import DownloadMultimediaZip
+from corehq.util.global_request import get_request
```
Comment (dimagimon) [inline corehq/apps/app_manager/views/download.py]: F401 'get_request' imported but unused

**Comment by dannyroberts** [comment]
I'll let you clean up the linter comments but looks good to me!

**Comment by nickpell** [comment]
Cleaned up imports.  The remaining linter errors in corehq/apps/app_manager/views/__init__.py are intentional since that file consists entirely of imports

**Comment by dannyroberts** [comment]
Ah, ok.

**Comment by NoahCarnahan** [comment]
The bug that caused reverting, for reference: http://manage.dimagi.com/default.asp?190622

**Pull request merged**